saleor/payment/gateways/braintree/__init__.py

```diff
@@ -137,127 +115,85 @@ def get_client_token(**connection_params: Dict) -> str:
 
 
 def authorize(
-        payment: Payment,
+        payment: Dict,
         payment_token: str,
-        **connection_params: Dict) -> Tuple[Transaction, str]:
+        **connection_params: Dict) -> Tuple[bool, Dict, str]:
     gateway = get_braintree_gateway(**connection_params)
+
     try:
         result = gateway.transaction.sale({
-            'amount': str(payment.total),
+            'amount': str(payment['total']),
             'payment_method_nonce': payment_token,
             'options': {
                 'submit_for_settlement': CONFIRM_MANUALLY,
                 'three_d_secure': {
                     'required': THREE_D_SECURE_REQUIRED}},
             **get_customer_data(payment)})
     except braintree_sdk.exceptions.NotFoundError:
-        return transaction_and_incorrect_token_error(
-            payment, kind=TransactionKind.AUTH, token=payment_token)
+        # FIXME: make this a payment generic exception
+        raise Exception('Unable to process the transaction. '
+            'Transaction\'s token is incorrect or expired.')
```

Comment:
Currently not, I was thinking what's the best approach here if the provider package throws an exception and my thinking was that the Saleor gateway implementation would handle those and re-raise exception to Saleor if need be. I did not look into specifics when Braintree raises an exception and when it returns just errors with `is_success = False`, but I guess returning instead of re-raising would serve the same purpose..